xmppserver/src/main/java/org/jivesoftware/openfire/group/AbstractGroupProvider.java

```diff
@@ -95,8 +95,7 @@ public void addMember(String groupName, JID user, boolean administrator) throws
      * @throws UnsupportedOperationException if the provider is read only
      */
     @Override
-    public void updateMember(String groupName, JID user, boolean administrator)
-    {
+    public void updateMember(String groupName, JID user, boolean administrator) throws GroupNotFoundException {
```

Comment:
We're changing the 'contract' of this method, by having it throw GroupNotFoundException. We should add that to the interface, so that we force all implementations to adhere to that.

Reply:
... which you did. Sorry! :)